tests/framework/integration.go

```diff
@@ -155,3 +157,38 @@ func (c integrationClient) Compact(rev int64, o config.CompactOption) (*clientv3
 	}
 	return c.Client.Compact(ctx, rev, clientOpts...)
 }
+
+func (c integrationClient) Status() ([]*clientv3.StatusResponse, error) {
+	endpoints := c.Client.Endpoints()
+	var resp []*clientv3.StatusResponse
+	for _, ep := range endpoints {
+		status, err := c.Client.Status(context.Background(), ep)
+		if err != nil {
+			return nil, err
+		}
+		resp = append(resp, status)
+	}
+	return resp, nil
+}
+
+func (c integrationClient) HashKV(rev int64) ([]*clientv3.HashKVResponse, error) {
+	endpoints := c.Client.Endpoints()
+	var resp []*clientv3.HashKVResponse
+	for _, ep := range endpoints {
+		hashKV, err := c.Client.HashKV(context.Background(), ep, rev)
+		if err != nil {
+			return nil, err
+		}
+		resp = append(resp, hashKV)
+	}
+	return resp, nil
+}
+
+func (c integrationClient) Health() error {
+	_, err := c.Client.Get(context.Background(), "health")
```

Comment:
It 's  migrating `ctl_v3_endpoint_test.go` to common. Should we follow logic with `ctl_v3_endpoint_test` ?
https://github.com/kkkkun/etcd/blob/dcd0d3fc9cf38c2e79f4179494c9b71351ecd3e1/tests/e2e/ctl_v3_endpoint_test.go#L42

Reply:
When migrating `ctl_v3_endpoint_test.go`  to comon. I  think it's ok. 
If health logic needs to add a grpc check. I will fix by other PR.  
WDYT?

Reply:
Not sure I understand, e2e tests should follow it's logic and call `etcdctl`, integration tests should follow it's logic and use GRPC check.

Reply:
Just to be more clearer on top of @serathius 's comment. e2e test is simulating users' behavior, both the etcd cluster and client should get started from real binaries (`etcd` or `etcdctl`). 

While integration test doesn't depend on binaries at all, it just calls functions/methods directly to start etcd cluster, and uses client v3 instead of etcdctl to communicating with the cluster.

Reply:
That's ok.   I will add grpc health check.